### PR TITLE
fix($compile): `bindToController` should work without `controllerAs`

### PR DESCRIPTION
**Pull request by thorn0**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

fix 

**What is the current behavior? (You can also link to an open issue here)**

see #15088

**Does this PR introduce a breaking change?**

no

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by Narretz** [comment]
Looks like we can decide between this and 15106

**Comment by thorn0** [comment]
Yep, I didn't see #15106

**Comment by gkalpak** [comment]
Let's keep this one. It's more complete than #15106.

**Comment by thorn0** [comment]
Is there a way to restart Travis without pushing a new commit?

**Comment by gkalpak** [comment]
There is :smile:

File: test/ng/compileSpec.js [hunk 1]
```diff
@@ -6126,154 +6126,114 @@ describe('$compile', function() {
...
+                                      'dir-str="Hello, {{whom}}!" ' +
+                                      'dir-fn="fn()"></div>')($rootScope);
+                    $rootScope.$digest();
+                    expect(controllerCalled).toBe(true);
+                    if (ddo.controllerAs || ddo.controller.indexOf(' as ') !== -1) {
+                      if (ddo.scope) {
+                        expect($rootScope.myCtrl).toBeUndefined();
+                      } else {
+                        // The controller identifier was added to the containing scope!
+                        // Should we allow this to happen?
```
Comment (gkalpak) [inline test/ng/compileSpec.js]: Why not? It's their code, they are allowed to do things :smiley:
In any case, this is not related to this PR. Remove this comment and feel free to open a new issue about it if you feel it shouldn't happen.

**Comment by gkalpak** [comment]
LGTM as long as Travis is happy (and unless someone can think of a good reason not to allow this).

File: test/ng/compileSpec.js (hunk 1 above)
Comment (thorn0) [inline test/ng/compileSpec.js]: Just wasn't sure about it...

**Comment by Narretz** [comment]
LGTM, please merge

**Pull request closed without merging**